### DISCUSSION TITLE
Need Assistance in Retrieving Parent GUIDs for IfcSpatialElement Entities in IFC4 File Parsing Script

### DISCUSSION THREAD

**Discussion by ashi7upt** [IfcOpenShell/IfcOpenShell#3618] [Q&A] [answered]
I am working on a script to assign parent GroupIDs to various IFC entities within an IFC4 file. I have made progress in assigning GroupIDs to the "IfcProject" subclass. However, I'm encountering difficulties when trying to obtain the parent GUIDs for "IfcSpatialElement" entities.

Here's a simplified version of my current code:

def build_ifc_relationship_map(ifc_file):
    rel_map = {}
    for rel in ifc_file.by_type("IfcRelationship"):
        if hasattr(rel, "RelatedObjects") and hasattr(rel, "RelatingObject"):
            parent = rel.RelatingObject
            children = rel.RelatedObjects
            for child in children:
                rel_map[child] = parent
    return rel_map
My question is: How can I modify this code to successfully retrieve the parent GUIDs for "IfcSpatialElement" entities? I've been able to obtain the parent GroupID for the "IfcProject" subclass, but I'm unsure about the approach to retrieve the parent GUIDs for "IfcSpatialElement" entities. Any guidance or suggestions on how to achieve this would be greatly appreciated.

Thank you in advance for your assistance!

**Comment by Moult** [comment]
```python
for element in ifc_file.by_type("IfcSpatialElement"):
    parent = ifcopenshell.util.element.get_aggregate(element)
    print(parent.GlobalId)
```

**Reply by ashi7upt** [reply]
Thank you but this code gives you the Groupid till IFCspace I also need parent Groupids for wall and windows

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/sahrm/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/sahrm/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



175089fc-d046-481f-98dd-a0fb87f9fd73 | IfcPlate | 2be31996-d1f1-4a9e-800b-2fb45d0acae9
-- | -- | --
72afb6b0-4caf-4b64-a14d-4fcb942871f0 | IfcPlate | 2be31996-d1f1-4a9e-800b-2fb45d0acae9
924d0898-ae72-4f83-a925-b8254178909a | IfcDoor |  
96473dbb-96f1-484c-94d9-1247cb11abc6 | IfcWindowType |  
be8ada01-697b-498d-9f7d-eda85c640a22 | IfcWindow |  
36c71924-3c8d-4e01-9a30-0063c219153e | IfcWindow |  
c6a2a025-c1df-4c5d-b9d7-1a99bf00dac5 | IfcWindow |  
b2997815-5024-458c-943f-52b1961a34b6 | IfcWindow |  
1b126216-a5f2-479a-95aa-0dc5e90add90 | IfcWindow |  
3967730b-ae8b-4b1f-806c-b5c437ae3cd4 | IfcWindow |  
298da3e7-d727-4ab6-8753-b8e7c95a8529 | IfcOpeningElement |  



</body>

</html>

**Reply by Moult** [reply, marked as answer]
Do you mean you want to know the GlobalId of the spatial element that an object (e.g. a wall/ window) is in?

```python
for element in ifc_file.by_type("IfcWall"):
    parent = ifcopenshell.util.element.get_container(element)
    print(parent.GlobalId)
```

**Reply by ashi7upt** [reply]
yes, I want to know these but they are not coming.

**Reply by Moult** [reply]
I just gave you the code that does what you're after (the `get_container` function).

**Reply by ashi7upt** [reply]
thank you
